src/main.rs

```diff
@@ -71,45 +72,62 @@ fn encode(words: &[String]) -> (Vec<u32>, HashMap<u32, Vec<String>>) {
 fn encode_word(word: &str) -> u32 {
     let mut code: u32 = 0;
     for char in word.chars() {
-        code |= 1 << (char as u32 - 97);
+        assert!(char.is_ascii_alphabetic());
+        let ucode = char.to_ascii_lowercase() as u32;
+        code |= 1 << (ucode - 'a' as u32);
     }
     code
 }
 
-fn solve_outer(codes: &Vec<u32>) -> Vec<Solution> {
-    (0..codes.len())
-        .into_par_iter()
-        .map(|idx| solve_inner(codes, idx, 0, 4))
-        .flatten()
+fn solve_outer(codes: &[u32]) -> Vec<Solution> {
+    #[cfg(feature = "rayon")]
+    let c_iter = (0..codes.len()).into_par_iter();
```

Comment:
I wanted to add an option for a user to specify how many parallel processes / threads / ... (I didn't find information in the docs about that) should be used. Imagine a `-j8` flag in a CLI. I'm guessing that is not possible with `rayon` right? If yes, I think this "boolean" behavior is good enough.

Reply:
You could add an argument parser that looks for a runtime flag for number of threads in the `cfg(feature = "rayon")` case. Rayon allows you to set that up with its threadpool builder. You could also just let the user use the `RAYON_NUM_THREADS` environment variable: less cool but should work. https://stackoverflow.com/questions/59205184/how-can-i-change-the-number-of-threads-rayon-uses